src/DynamoCoreWpf/Views/Search/LibrarySearchView.xaml.cs

```diff
@@ -177,7 +183,20 @@ private void ShowTooltip(object sender)
             if (fromSender == null) return;
 
             libraryToolTipPopup.PlacementTarget = fromSender;
-            libraryToolTipPopup.SetDataContext(fromSender.DataContext);
+            if ((fromSender.DataContext as NodeSearchElementViewModel).Visibility)
+                libraryToolTipPopup.SetDataContext(fromSender.DataContext);
+            else
+                CloseTooltip();
+        }
+
+        private void CloseTooltip(bool closeImmediately = false)
+        {
+            libraryToolTipPopup.SetDataContext(null, closeImmediately);
+        }
+
+        private void CloseTooltip(object sender, EventArgs e)
+        {
+            libraryToolTipPopup.SetDataContext(null, true);
```

Comment:
I think your idea was to call `CloseToolTip(true)` instead of `libraryToolTipPopup.SetDataContext` here (so that we only deal with `SetDataContext` in one place)?